src/hooks/chart.ts

```diff
@@ -370,7 +365,7 @@ export const useChartData = (plot: Plot) => {
         if (trace.url_pattern || trace.queryPattern) {
           //To avoid the lint error, passing empty string if both are not present'
           const pattern = trace.url_pattern || trace.queryPattern || '';
-          const { uri, headers } = getPatternUri(pattern, templateParams);
+          const { uri, headers } = getPatternUri(pattern, useInitialParams ? initialParams: templateParams);
```

Comment:
Why does this need to be done?

Since we are initializing everything in `plot-control-grid`, templateParams from the provider should be available. There shouldn't be a prop passed to `useChartData` hook for the "initial params". Instead `useChartData` hook should be setup only when the `templateParams` have been initialized in the provider.